src/constructs/iam/policies/s3-get-object.ts

```diff
@@ -1,26 +1,25 @@
-import { Effect, PolicyStatement } from "@aws-cdk/aws-iam";
 import type { GuStack } from "../../core";
+import { GuSSMParameter } from "../../core";
 import type { GuPolicyProps } from "./base-policy";
-import { GuPolicy } from "./base-policy";
+import { GuAllowPolicy } from "./base-policy";
 
 export interface GuGetS3ObjectPolicyProps extends GuPolicyProps {
   bucketName: string;
 }
 
-export const allowGetObjectPolicyStatement: (bucket: string) => { statements: PolicyStatement[] } = (
-  bucketName: string
-) => ({
-  statements: [
-    new PolicyStatement({
-      effect: Effect.ALLOW,
-      actions: ["s3:GetObject"],
-      resources: [`arn:aws:s3:::${bucketName}/*`],
-    }),
-  ],
-});
-
-export class GuGetS3ObjectPolicy extends GuPolicy {
+export class GuGetS3ObjectPolicy extends GuAllowPolicy {
   constructor(scope: GuStack, id: string, props: GuGetS3ObjectPolicyProps) {
-    super(scope, id, { ...allowGetObjectPolicyStatement(props.bucketName), ...props });
+    super(scope, id, { ...props, actions: ["s3:GetObject"], resources: [`arn:aws:s3:::${props.bucketName}/*`] });
+  }
+}
+
+export class GuGetDistributablePolicy extends GuGetS3ObjectPolicy {
+  constructor(scope: GuStack, id: string = "GetDistributablePolicy", props?: GuPolicyProps) {
```

Comment:
Maybe this should be `GuNoStatementsPolicyProps`? And should `GuGetS3ObjectPolicyProps` extend that instead too?

Reply:
Yeah wasn't sure what to use here. I don't think `GuGetS3ObjectPolicyProps` would be correct as it has a `bucketName` property, which we don't want the user to set as it comes from the parameter.

Reply:
I've done something along the lines of this elsewhere which might work here too

```ts
const json = SynthUtils.toCloudFormation(stack) as SynthedStack;

expect(json.Parameters.DistributionBucketName).toEqual({
  Default: "/account/services/artifact.bucket",
  Description: "SSM parameter containing the S3 bucket name holding distribution artifacts",
  NoEcho: true,
  Type: "AWS::SSM::Parameter::Value<String>",
});
```

Reply:
We could also remove `props` entirely? The only thing we're currently passing through is [`policyName`](https://github.com/guardian/cdk/blob/d240cb1587ec02d0ab746f9c72403e185ba380b0/src/constructs/iam/policies/s3-get-object.test.ts#L30) and I think the autogenerated name would be fine here.

Reply:
That's true... although if we do that, let's do it everywhere for consistency. I'm somewhat hesitant about removing it but I don't actually have any valid reasons. As you say, `policyName` is the only other prop that we've used so far and I agree that the auto-generate name should be fine.

Reply:
Lets go with `GuNoStatementsPolicyProps` for now then.